tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_connect.c

```diff
@@ -42,6 +42,18 @@ static const struct bt_data ad[] = {
 };
 
 static bool volatile is_connected;
+static int volatile num_eatt_channels;
+
+static void eatt_chan_connected(struct bt_att_chan_info *info)
+{
+	printk("ATT channel connected. cid: 0x%04X, mtu: %d. mps: %d\n", info->cid, info->mtu,
+	       info->mps);
+
+	num_eatt_channels++;
+	if (num_eatt_channels > CONFIG_BT_EATT_MAX) {
+		FAIL("Too many EATT channels connected (%d)\n", num_eatt_channels);
+	}
```

Comment:
Given that the peripheral also has the same `CONFIG_BT_EATT_MAX`, shouldn't the check have been

```suggestion
	if (num_eatt_channels != CONFIG_BT_EATT_MAX) {
		FAIL("Incorrect EATT channels connected %u/%u\n", num_eatt_channels, CONFIG_BT_EATT_MAX);
	}
```

Reply:
This callback is called for each EATT channel. We can thus not fail the test the first time the callback is called and `num_eatt_channels==1`.

Can update the message to include `CONFIG_BT_EATT_MAX`.

Reply:
I see. Do we fail the test if we don't connect `CONFIG_BT_EATT_MAX` channels somewhere else then?

Reply:
If we connect too few, the test will time out as the condition on line 131 and 198 never becomes false.
```c
while (num_eatt_channels < CONFIG_BT_EATT_MAX) {
```